doc/test_organization/managing_tests_dependencies.qbk

```diff
@@ -7,17 +7,31 @@
 
 [section:tests_dependencies Managing test dependencies]
 
-Dependencies among tests are managed by means of [link boost_test.tests_organization.decorators decorators]. 
+In general, you should make sure that test cases are independent on one another.
```

Comment:
I did not have a chance to read through this. Here few points I'd like to
make:


   1. Default order follows order of test unit declaration in translation
   unit
   2. There is no promise bet ween translation units
   3. All test units within the same test suite are executed together
   4. Dependency enforces order, not the other way around
   5. random test unit execution shuffles test siblings of *all* test
   suites independently.
   6. Most complicated to explain to the case of dependency + random. What
   happens is this: while analyzing the dependency we assign "rank" to all
   siblings among test suite children. Lets say we have 2 children with rank 1
   - tu1 and tu2 and 2  children with rank 2 - tu3 and tu4. We will ran first
   tu1 and tu2 in random order, followed by tu3 and tu4 in random order


On Mon, May 25, 2015 at 4:32 AM, Raffi Enficiaud <notifications@github.com>
wrote:

> In doc/test_organization/managing_tests_dependencies.qbk
> <https://github.com/boostorg/test/pull/64#discussion_r30968769>:
>
> > @@ -7,17 +7,31 @@
> >
> >  [section:tests_dependencies Managing test dependencies]
> >
> > -Dependencies among tests are managed by means of [link boost_test.tests_organization.decorators decorators].
> > +In general, you should make sure that test cases are independent on one another.
>
> They can be ordered without dependency, aren't they? Dependency enforce
> order, the opposite is not true.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/boostorg/test/pull/64/files#r30968769>.
>



-- 
Gennadiy Rozental